lib/auth0_client.rb

```diff
@@ -0,0 +1,43 @@
+# frozen_string_literal: true
+
+require 'jwt'
+require 'net/http'
+
+class Auth0Client
+  Auth_Domain = URI::HTTPS.build(host: Rails.application.config.x.auth0.domain)
+  Error = Struct.new(:message, :status)
+  Response = Struct.new(:decoded_token, :error)
+
+  def self.decode_token(token, jwks_hash)
+    JWT.decode(token, nil, true, {
+                 algorithm: 'RS256',
+                 iss: URI.join(Auth_Domain, '/'), # Trailing slash is required to successfully validate token
+                 verify_iss: true,
+                 aud: Rails.application.config.x.auth0.audience,
+                 verify_aud: true,
+                 jwks: { keys: jwks_hash[:keys] }
+               })
+  end
+
+  def self.fetch_jwks
+    jwks_uri = URI.join(Auth_Domain, '.well-known/jwks.json')
+    Net::HTTP.get_response jwks_uri
+  end
+
+  def self.validate_token(token)
+    jwks_response = fetch_jwks
+
+    unless jwks_response.is_a? Net::HTTPSuccess
+      error = Error.new(message: 'Unable to verify credentials', status: :internal_server_error)
+      Response.new(nil, error)
+    end
+
+    jwks_hash = JSON.parse(jwks_response.body).deep_symbolize_keys
+    decoded_token = decode_token(token, jwks_hash)
+    Response.new(decoded_token, nil)
+  # rubocop:disable Lint/ShadowedException
+  rescue JWT::DecodeError, JWT::VerificationError
```

Comment:
I think the Rubocop error is because `JWT::VerificationError` is a subclass of `JWT::DecodeError`, which means that it's redundant to specify `JWT::VerificationError` because `JWT::DecodeError` will already handle it.

Maybe remove the redundant error, which should allow you to remove the `rubocop:disable` comment?

Reply:
Ahh thanks! That was stumping me